nestjs-counter/apps/worker/src/temporal/workflows.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { Runtime, DefaultLogger, Worker } from '@temporalio/worker';
+import { WorkflowClient, WorkflowHandle } from '@temporalio/client';
+import { getExchangeRatesQuery } from '@app/shared';
+import { exchangeRatesWorkflow } from './workflows';
+
+const taskQueue = 'test-exchange-rates';
+
+describe('example workflow', function () {
+  let client: WorkflowClient;
+  let handle: WorkflowHandle;
+  let shutdown: () => Promise<void>;
+  let execute: () => Promise<WorkflowHandle>;
+  let env: TestWorkflowEnvironment;
+
+  beforeAll(async function () {
+    const activities = {
+      getExchangeRates: () => Promise.resolve({ AUD: 1.27 }),
+    };
+    jest.spyOn(activities, 'getExchangeRates');
+
+    // Filter INFO log messages for clearer test output
+    Runtime.install({ logger: new DefaultLogger('WARN') });
+    env = await TestWorkflowEnvironment.create();
+    const worker = await Worker.create({
+      connection: env.nativeConnection,
+      taskQueue,
+      workflowsPath: require.resolve('./workflows.ts'),
+      activities,
+    });
```

Comment:
I'd show how to integrate the test environment with nestjs so you can use the worker provider instead of rewriting all of this in the test.

Reply:
Comment still applies.